### ISSUE TITLE
Adyen checkout page uses site language instead of preferred user language

### ISSUE THREAD

**Issue by rvdsteege** [pronamic/wp-pronamic-pay#136]
The Adyen checkout page uses the site language for logged in users, instead of the language setting from the user profile settings.

- https://github.com/wp-pay-gateways/adyen/blob/1.3.1/views/checkout-drop-in.php

**Comment by remcotolsma** [comment]
I noticed that there is a `determine_locale` function that determines which locale to use:

https://github.com/WordPress/WordPress/blob/058f9903676a7efaee534a682df0a2a8b87574d8/wp-includes/l10n.php#L112-L159

In the WordPress admin dashboard it will use the user locale:

```php
if ( is_admin() ) {
	$determined_locale = get_user_locale();
}
```

If we also want this for the Adyen checkout pages we could add something similar via the `'determine_locale'` filter?

**Comment by rvdsteege** [comment]
Thanks for the research, @remcotolsma!

I later noticed the same issue occurs with our regular payment redirect (when performed from the frontend, not when testing a configuration — because of the `is_admin()` check) and possibly also on other pages.

I'll self-assign this issue, to improve using the user locale if possible.

**Comment by remcotolsma** [comment]
Maybe not required for the 6.8 milestone? Then we might be able to resolve the already existing 6.8 issues and launch 6.8 this week.

**Comment by rvdsteege** [comment]
I'll try to fix this for `6.8`, but otherwise we can postpone this one indeed.
